src/main/java/de/zalando/aruha/nakadi/config/WebConfig.java

```diff
@@ -45,18 +47,17 @@ public TimeoutCallableProcessingInterceptor timeoutInterceptor() {
 
     @Bean
     public FilterRegistrationBean flowIdRequestFilter() {
-        final FilterRegistrationBean filterRegistrationBean = new FilterRegistrationBean();
-        filterRegistrationBean.setFilter(new FlowIdRequestFilter());
-        filterRegistrationBean.setOrder(Ordered.HIGHEST_PRECEDENCE + 1);
-        return filterRegistrationBean;
+        return createFilterRegistrationBean(new FlowIdRequestFilter(), Ordered.HIGHEST_PRECEDENCE + 2);
```

Comment:
Me, personally, I think I would execute the FlowIdRequestFilter first and the GzipBodyRequestFilter afterwards. If something during the decompression goes wrong, you already have the FlowId in the logs.